tests/testthat/test-duplicate_argument_linter.R

```diff
@@ -0,0 +1,65 @@
+test_that("returns the correct linting", {
+  expect_lint("fun(arg = 1)",
+    NULL,
+    duplicate_argument_linter())
+
+  expect_lint("fun('arg' = 1)",
+    NULL,
+    duplicate_argument_linter())
+
+  expect_lint("fun(`arg` = 1)",
+    NULL,
+    duplicate_argument_linter())
+
+  expect_lint("'fun'(arg = 1)",
+    NULL,
+    duplicate_argument_linter())
+
+  expect_lint("(function(x, y) x + y)(x = 1)",
+    NULL,
+    duplicate_argument_linter())
+
+  expect_lint("dt[i = 1]",
+    NULL,
+    duplicate_argument_linter())
+
+  expect_lint("fun(arg = 1, arg = 2)",
+    list(message = rex("Duplicate arguments in function call.")),
+    duplicate_argument_linter())
+
+  expect_lint("fun(arg = 1, 'arg' = 2)",
+    list(message = rex("Duplicate arguments in function call.")),
+    duplicate_argument_linter())
+
+  expect_lint("fun(arg = 1, `arg` = 2)",
+    list(message = rex("Duplicate arguments in function call.")),
+    duplicate_argument_linter())
+
+  expect_lint("'fun'(arg = 1, arg = 2)",
+    list(message = rex("Duplicate arguments in function call.")),
+    duplicate_argument_linter())
+
+  expect_lint("(function(x, y) x + y)(x = 1, x = 2)",
+    list(message = rex("Duplicate arguments in function call.")),
+    duplicate_argument_linter())
+
+  expect_lint("dt[i = 1, i = 2]",
+    list(message = rex("Duplicate arguments in function call.")),
+    duplicate_argument_linter())
+
+  expect_lint(
+    "list(
+      var = 1,
+      var = 2
+    )",
+    list(message = rex("Duplicate arguments in function call.")),
+    duplicate_argument_linter())
+
+  expect_lint(
+    "list(
+      var = 1,
+      var = 2
+    )",
+    NULL,
+    duplicate_argument_linter(except = "list"))
```

Comment:
Can we test the `except = ...` branch with anonymous functions, too?
This ensures the `calls_names` are detected properly for calls to anonymous functions.

Reply:
Yes, exactly.